### DISCUSSION TITLE
What does num_boundary mean when using PointSetBC ?

### DISCUSSION THREAD

**Discussion by baichen99** [lululxvi/deepxde#1089] [Q&A] [answered]
Hi Dr. Lu

I use the data generated by openfoam as pointsetBC as input to the network, in this case, is num_boundary the point adopted from pointsetBC?

Also when I use PDEPointResampler, will it resample points from pointsetBC, why don't set `bc_points=True` in [this example](https://github.com/lululxvi/deepxde/blob/master/examples/pinn_inverse/elliptic_inverse_field_batch.py), I hope you can answer my question, thanks!

Here are part of my code:

```python
observe_y1 = dde.icbc.PointSetBC(X, u_x, component=0, batch_size=256)
observe_y2 = dde.icbc.PointSetBC(X, u_y, component=1, batch_size=256)
#...
data = dde.data.TimePDE(
    geomtime,
    pde,
    [observe_y1, observe_y2],
    num_domain=700,
    num_boundary=0,
    num_initial=0,
    anchors=X
)

# ...

# Compile, train and save model
model.compile("adam", lr=1e-3)

loss_history, train_state = model.train(
    iterations=EPOCHS, display_every=1000, disregard_previous_best=True,
    callbacks=[pde_resampler]
)
```

**Comment by haison19952013** [comment, marked as answer]
- Do you have any additional boundary condition function for your case? If not, `num_boundary` has no meaning and should be set to 0.
- As I read here: https://github.com/lululxvi/deepxde/blob/56cd98b241114ccc93e5dc8dc1bec761fced1b26/deepxde/callbacks.py#L540, it seems `bc_points` is only supported with Pytorch. And as I understand,  if you set `bc_points = True` , and in the case you have already defined a boundary condition function, the points on the BCs will be resampled.

**Reply by baichen99** [reply]
It solved my problem, thanks for the answer!
